File: src-tauri/icons/generate-icons.js
/**
 * Simple icon generator for PoliGame
 * Requires: npm install sharp (or use online converter)
 */

const fs = require('fs');
const path = require('path');

console.log('PoliGame Icon Generator');
console.log('======================\n');

// Check if sharp is available
let sharp;
try {
  sharp = require('sharp');
  console.log('Sharp found! Generating icons from SVG...\n');
  
  const svgPath = path.join(__dirname, 'icon.svg');
  
  // Generate PNG icons
  async function generateIcons() {
    try {
      // 32x32
      await sharp(svgPath)
        .resize(32, 32)
        .png()
        .toFile('32x32.png');
      console.log('✓ Generated 32x32.png');
      
      // 128x128
      await sharp(svgPath)
        .resize(128, 128)
        .png()
        .toFile('128x128.png');
      console.log('✓ Generated 128x128.png');
      
      // 256x256 (128x128@2x)
      await sharp(svgPath)
        .resize(256, 256)
        .png()
        .toFile('128x128@2x.png');
      console.log('✓ Generated 128x128@2x.png');
      
      // ICO (Windows) - sharp doesn't support ICO directly, so we'll create a simple one
      // For proper ICO, use online converter or ImageMagick
      console.log('\n⚠ ICO file generation requires ImageMagick or online converter');
      console.log('   Use: https://convertio.co/png-ico/');
      console.log('   Or: magick icon.svg -resize 256x256 icon.ico\n');
      
      console.log('✓ PNG icons generated successfully!');
    } catch (error) {
      console.error('Error generating icons:', error.message);
    }
  }
  
  generateIcons();
  
} catch (e) {
  console.log('Sharp not found. Installing...');
  console.log('Run: npm install sharp\n');
  console.log('Or use one of these methods:\n');
  console.log('1. Python script (if Python installed):');
  console.log('   pip install cairosvg pillow');
  console.log('   python generate-icons.py\n');
  console.log('2. PowerShell script (if ImageMagick installed):');
  console.log('   powershell -ExecutionPolicy Bypass -File generate-icons.ps1\n');
  console.log('3. Online converter:');
  console.log('   https://cloudconvert.com/svg-to-png');
  console.log('   Upload icon.svg and convert at sizes: 32, 128, 256\n');
}

File: src-tauri/icons/generate-icons.py
#!/usr/bin/env python3
"""
Generate PoliGame app icons from SVG
Requires: pip install cairosvg pillow
"""

try:
    import cairosvg
    from PIL import Image
    import io
    import os
except ImportError:
    print("Required packages not installed!")
    print("Install with: pip install cairosvg pillow")
    exit(1)

def generate_icon(size, output_path):
    """Generate PNG icon from SVG at specified size"""
    svg_path = "icon.svg"
    
    # Convert SVG to PNG
    png_data = cairosvg.svg2png(url=svg_path, output_width=size, output_height=size)
    
    # Save PNG
    with open(output_path, 'wb') as f:
        f.write(png_data)
    
    print(f"Generated {output_path} ({size}x{size})")

def generate_ico():
    """Generate ICO file with multiple sizes"""
    sizes = [256, 128, 64, 48, 32, 16]
    images = []
    
    svg_path = "icon.svg"
    
    for size in sizes:
        png_data = cairosvg.svg2png(url=svg_path, output_width=size, output_height=size)
        img = Image.open(io.BytesIO(png_data))
        images.append(img)
    
    # Save as ICO
    images[0].save("icon.ico", format='ICO', sizes=[(s, s) for s in sizes])
    print(f"Generated icon.ico with sizes: {', '.join(map(str, sizes))}")

def main():
    print("PoliGame Icon Generator")
    print("======================\n")
    
    # Generate PNG icons
    generate_icon(32, "32x32.png")
    generate_icon(128, "128x128.png")
    generate_icon(256, "128x128@2x.png")
    
    # Generate ICO
    generate_ico()
    
    print("\n✓ All icons generated successfully!")
    print("\nNote: For macOS .icns file, you may need to use iconutil or online converter")

if __name__ == "__main__":
    main()

